### PR TITLE
Add Toit syntax highlight

### PR DESCRIPTION
**Pull request by snxx-lppxx**
- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [x] My package doesn't add context menu entries. *
- [x] My package doesn't add key bindings. **
- [x] Any commands are available via the command palette.
- [x] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [x] If my package is a syntax it doesn't also add a color scheme. ***
- [ ] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

Added syntax highlighting for the Toit programming language.

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

### CODE REVIEW

**Comment by braver** [comment]
Please have a look at the feedback from the bot.

**Comment by jrappen** [comment]
* The `uuid` key in the syntax file is not valid (i.e. not a known key)
* The `comment` key in the syntax file is also not valid (i.e. not a known key).
* You might want to run [tests](https://www.sublimetext.com/docs/syntax.html#testing), compare `./.github/workflows/` and the `**/syntax_test_*` files in <https://github.com/sublimehq/Packages>
* You can use the `*.gitattributes` file to **not** ship certain files with your package, random example at <https://github.com/sublimelsp/LSP>
* You included screenshots in the repo, but do not link them in the README. Why?
* The `DOCTYPE` line in (the) `tmPreferences` file(s) is redundant.
* The uuid in the `tmPreferences` file is unused (via Python from your package), hence redundant.
* Setting a `name` in the `tmPreferences` file is also redundant, as you copied the file name.
* Using 4 spaces indentation for JSON-like formatted files for Sublime core support is considered convention at this point AFAIK.
* Separating array items in JSON arrays by new line makes future diffs easier to read.
* ⚠️ I believe you cannot `extend` the syntax file from itself?
* ⚠️ Do not use capturing groups in regexes for variables. Use non-capturing groups there and surround references in contexts with the capturing group.
* ⚠️ The way you defined escape characters is wrong. Compare examples in the `Packages` repo linked above.
* ⚠️ The use of longer regexes to match a list of possible elements **might** lead to a reduced performance of your syntax compared to trying to match possible elements one-by-one (compare your regexes for string escape chars and operators for instance). Running `Tools > Build with ... > Syntax Test - Performance` from the menu in Sublime will give results.

---

... I ran out of time reviewing your stuff. Please find and fix remaining issues yourself.

**Comment by braver** [comment]
Please respond to the feedback to continue the PR.

File: repository/t.json [hunk 1]
```diff
@@ -2778,6 +2778,18 @@
 				}
 			]
 		},
+		{
+			"name": "Toit Syntax Highlighting",
```
Comment (braver) [inline repository/t.json:2782]: Syntax packages do more than simply highlighting, they enable various other features as well. Hence, language support packages should simply be named after the language. E.g. “css” and not “css syntax” or “css highlighting “.

**Review by braver — CHANGES_REQUESTED** [review]

**Comment by snxx-lppxx** [comment]
Hi there! I decided to change the version for Sublime. I don't have support for TextMate yet.

**Comment by snxx-lppxx** [comment]
Hi @braver, please review this.

**Comment by braver** [comment]
Please have a look at the report from the bot first.

**Comment by braver** [comment]
You should also be able to add `screenshots` and `tests` to the gitattributes. Otherwise it's fine,  let's ship!

**Pull request merged**